src/activities/menu/Menu.qml

```diff
@@ -47,6 +47,9 @@ ActivityBase {
     id: activity
     focus: true
     activityInfo: ActivityInfoTree.rootMenu
+    // set to true when returned to home from a dialog.
+    property bool returnFromDialog: false
```

Comment:
This variable is needed to handle the cases when we close a dialog.
home() is called when we close a dialog, so to prevent the confirmation dialog to pop up, in that case, I have used this variable to handle this.